technical-docs/designs/services/acrBuildArguments.md

```diff
@@ -0,0 +1,91 @@
+# Software Design Document: Passing arguments to a Dockerfile during `az acr build` execution
+
+Reference: An option to pass environment variables while building a Dockerfile
+<br> Authors: Bhargav Nookala, Evan Louie, James Spring, Michael Tarng, Yvonne
+Radsmikham
+
+| Revision | Date         | Author            | Remarks       |
+| -------: | ------------ | ----------------- | ------------- |
+|      0.1 | Mar-26, 2020 | Yvonne Radsmikham | Initial Draft |
+
+## 1. Overview
+
+This design will enhance the developer experience for using
+`spk service create`. Currently, the concern is that building of the Dockerfile
+assumes there are no inputs into the build process. This is handled in a script
+task of the `build` stage in the `build-update-hld.yaml`, which is generated
+when `spk service create` is executed.
+
+Snippet of `build-update-hld.yaml`:
+
+```yaml
+    - script: |-
+        set -e
+        export BUILD_REPO_NAME=$(echo $(Build.Repository.Name)-fabrikam.acme.frontend | tr '[:upper:]' '[:lower:]')
+        export IMAGE_TAG=$(echo $(Build.SourceBranchName) | tr / - | tr . - | tr _ - )-$(Build.BuildNumber)
+        export IMAGE_NAME=$BUILD_REPO_NAME:$IMAGE_TAG
+        echo "Image Name: $IMAGE_NAME"
+        cd ./services/fabrikam.acme.frontend
+        echo "az acr build -r $(ACR_NAME) --image $IMAGE_NAME ."
+        az acr build -r $(ACR_NAME) --image $IMAGE_NAME .
+    displayName: ACR Build and Publish
+```
+
+## 2. Out of Scope
+
+This design shall only aim to make user experience better by adding new features
+to service creation.
+
+## 3. Design Details
+
+### 3.1 Add new arguments for `spk service create` to link or create variable groups
+
+As part of `spk service create`, add the following features:
+
+- Create a new command argument `--build-vg` to link an existing variable group
+  with variables to be passed into a Docker build execution. An example would be
+
+```sh
+spk service create "fabrikam.acme.frontend" -d "services" -p "chart" -g "https://dev.azure.com/bedrock/fabrikam/_git/fabrikam2019" -b "master" --build-vg "my-azdo-build-vg"
+```
+
+- Add argument `--build-variables` to take in an array of key-value pairs for
+  all variables to be used in a new variable group. An example would be
+
+```sh
+spk service create "fabrikam.acme.frontend" -d "services" -p "chart" -g "https://dev.azure.com/bedrock/fabrikam/_git/fabrikam2019" -b "master" --build-variables [version:0.0.1,path:$HOME]
```

Comment:
The `key:value` pair supports non secret type variable by default. Does it need to support secret variable type? If so, it needs to be extended `key:value:type`  or `key:value:isSecret`.

Reply:
That's a good point. I will update the docs.